integration_tests/test_solo_machine.py

```diff
@@ -0,0 +1,182 @@
+import json
+import os
+import platform
+from pathlib import Path
+
+import pytest
+import requests
+from pystarport import ports
+from pystarport.utils import interact
+
+from .utils import cluster_fixture, wait_for_block, wait_for_port
+
+pytestmark = pytest.mark.solomachine
+
+
+@pytest.fixture(scope="module")
+def cluster(worker_index, pytestconfig, tmp_path_factory):
+    "override cluster fixture for this test module"
+    try:
+        yield from cluster_fixture(
+            Path(__file__).parent / "configs/solo_machine.yaml",
+            worker_index,
+            tmp_path_factory.mktemp("data"),
+        )
+    finally:
+        pass
+
+
+def parse_output(output, return_json=True):
+    s = output.decode("utf-8")
+    s = s.replace("\u001b[0m", "")
+    print(s)
+    if return_json:
+        data = json.loads(s)
+        return data
+    else:
+        return s
+
+
+CRO_DECIMALS = 10 ** 8
+SOLO_HD_PATH = "m/44'/394'/0'/0/0"
+SOLO_ACCOUNT_PREFIX = "cro"
+SOLO_ADDRESS_ALGO = "secp256k1"
+SOLO_FEE_DENOM = "basecro"
+SOLO_DENOM = "solotoken"
+
+
+class SoloMachine(object):
+    def __init__(self, temp_path, base_port=26650, chain_id="devnet-solomachine"):
+        self.chain_id = chain_id
+        self.grpc_port = ports.grpc_port(base_port)
+        self.rpc_port = ports.rpc_port(base_port)
+        self.solomachine_home = os.path.join(os.environ["SOLO_MACHINE_HOME"])
+        self.bin_file = os.path.join(self.solomachine_home, "solo-machine")
+
+        os_platform = platform.system()
+        if os_platform == "Darwin":
+            self.sign_file = os.path.join(
+                self.solomachine_home, "libmnemonic_signer.dylib"
+            )
+        else:
+            self.sign_file = os.path.join(
+                self.solomachine_home, "libmnemonic_signer.so"
+            )
+        self.trusted_height = None
+        self.trusted_hash = None
+        self.db_path = f'sqlite://{temp_path.join("solo-machine.db")}'
+
+    def get_chain_info(self):
+        result = requests.get(f"http://127.0.0.1:{self.rpc_port}/block").json()
+        self.trusted_hash = result["result"]["block_id"]["hash"]
+        self.trusted_height = result["result"]["block"]["header"]["height"]
+
+    def set_env(self):
+        os.environ["SOLO_DB_URI"] = self.db_path
+        os.environ["SOLO_SIGNER"] = self.sign_file
+        os.environ[
+            "SOLO_MNEMONIC"
+        ] = "awesome there minute cash write immune tag \
+reopen price congress trouble reunion south wisdom donate credit below leave \
+wisdom eagle sail siege rice train"
```

Comment:
why duplicate it in the test code and not take it from the test config ("solo_machine.yaml")?

Reply:
I'll fix it.